bindings_flutter/test/v3_test.dart

```diff
@@ -0,0 +1,71 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:math';
+import 'dart:typed_data';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:web3dart/web3dart.dart';
+
+import 'package:xmtp_bindings_flutter/xmtp_bindings_flutter.dart';
+
+void main() {
+  var dbDir;
+  setUpAll(() async {
+    dbDir = Directory.systemTemp.createTempSync();
+    return libxmtpInit();
+  });
+  tearDownAll(() {
+    libxmtpDispose();
+    dbDir.deleteSync(recursive: true);
+  });
+  test(
+    'creating a client with consistent installations',
+    () async {
+      var wallet = EthPrivateKey.createRandom(Random.secure());
+      var dbPath = "${dbDir.path}/${wallet.address.hex}.db";
+      var encryptionKey =
+          U8Array32(Uint8List.fromList(List.generate(32, (index) => index)));
+      var walletSign = (text) async => wallet.signPersonalMessageToUint8List(
+          Uint8List.fromList(utf8.encode(text)));
+      var host = "http://localhost:5556";
+
+      // When we create a client...
+      var createdA = await createClient(
+        host: host,
+        isSecure: false,
+        dbPath: dbPath,
+        encryptionKey: encryptionKey,
+        accountAddress: wallet.address.hex,
+      );
+      // ... it should require a signature...
+      var clientA = switch (createdA) {
+        CreatedClient_Ready() => throw StateError("Should require a signature"),
+        CreatedClient_RequiresSignature(field0: var req) =>
+          await req.sign(signature: await walletSign(req.textToSign)),
+      };
```

Comment:
@richardhuaaa (in re [new signature flow](https://github.com/xmtp/libxmtp/pull/461#issuecomment-1918032981))

Here's how I exposed the flow (trying to keep it simple but still guide the app dev through how to use it properly):
```dart
var created = await createClient(...);
var client = switch (created) {
  CreatedClient_Ready(field0: var client) => client,
  CreatedClient_RequiresSignature(field0: var req) =>
    await req.sign(signature: await walletSign(req.textToSign)),
};
```
i.e. `createClient` returns a `CreatedClient` which is a sealed type that is either `Ready` or `RequiresSignature`. The `RequiresSignature` instance includes the `.textToSign` alongside a `.sign(...)` method that takes the `signature` and returns the ready `Client` upon success.

Reply:
on the rust side that looks like this (check `libxmtp_api.rs`):

```rs

pub enum CreatedClient {
    Ready(Client),
    RequiresSignature(SignatureRequiredClient),
}

pub struct SignatureRequiredClient {
    pub text_to_sign: String,
    pub inner: Arc<InnerClient>,
}
```